src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -185,7 +185,8 @@ void MeshForwarder::HandleResolved(const Ip6::Address &aEid, Error aError)
             hopLimit++;
             message.Write(Ip6::Header::kHopLimitFieldOffset, hopLimit);
 
-            IgnoreError(Get<Ip6::Ip6>().HandleDatagram(message, Ip6::Ip6::kFromHostAllowLoopBack));
+            message.SetOrigin(Message::Origin::kFromThreadNetif);
+            IgnoreError(Get<Ip6::Ip6>().HandleDatagram(message));
```

Comment:
It was `kFromHostAllowLoopBack` but the new origin is `kFromThreadNetif`. Is this intentional?

Reply:
Same question for me.

I think we should keep as it was before `kFromHostAllowLoopBack`?